provider.go

```diff
@@ -315,26 +342,27 @@ func (g *InstanceGroup) Increase(
 		})
 
 		if err != nil {
-			errs = append(errs, fmt.Errorf(
-				"failed to create %s: %w", serverName, err))
-			continue
-		}
-
-		server, err = g.client.Servers.WaitFor(
-			ctx, server.UUID, cloudscale.ServerIsRunning)
-
-		if err != nil {
-			errs = append(errs, fmt.Errorf(
-				"failed to wait for %s: %w", serverName, err))
+			errors_c <- fmt.Errorf("failed to create %s: %w", serverName, err)
 			continue
 		}
 
-		g.log.Info("created server", "name", serverName, "uuid", server.UUID)
-		servers = append(servers, server)
+		// Send the uuid to the "wait" goroutine
+		servers <- server.UUID
+	}
+	
+	// We wait for all previously started goroutines to complete before we return
+	wg.Wait()
+	// and we explicitly close the channel, so the waiting dispatcher
+	// (that for loop that recieves from the channel) can exit cleanly
+	close(servers)
+
+	// we collect all errors into a slice again
+	var errs []error
+	for err := range errors_c {
+		errs = append(errs, err)
 	}
 
 	return len(servers), errors.Join(errs...)
```

Comment:
The length here is zero if I'm not mistaken. I think you need to track the created servers in a separate variable and then return that. I'm not sure what GitLab Fleeting really expects when there are errors (does it just ignore the length of the servers created), but I think the best we can do is report accurately on the number of servers that we actually awaited (the await call can technically fail if a server cannot be scheduled, causing the server to first be launched, then discarded).

Reply:
Should be resolved by switching it back to a slice+mutex instead of the channel

Reply:
You are right, so i've added a counter instead